gpuweb/cts
# wgsl: compound statement tests

wgsl: Add compound statement tests
    
    Validation:
    - Check parsing of compound statements
    - Update break-if statement to add a case for a plain compound statement
      surrounding a break-if.
    - For a declaration in a compound statement, check potential uses
      and potentially conflicting declarations.
    
    Execution:
      Add execution tests for compound statement.
    
      Observe a value at various points relative to its
      declaration inside a compound statement.
    
    Add timings for new tests
    
Fixed: #1634
<hr>

**Requirements for PR author:**

- [x] All missing test coverage is tracked with "TODO" or `.unimplemented()`.
- [x] New helpers are `/** documented */` and new helper files are found in `helper_index.txt`.
- [x] Test behaves as expected in a WebGPU implementation. (If not passing, explain above.)

**Requirements for [reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md):**

- [ ] Tests are properly located in the test tree.
- [ ] [Test descriptions](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) allow a reader to "read only the test plans and evaluate coverage completeness", and accurately reflect the test code.
- [ ] Tests provide complete coverage (including validation control cases). **Missing coverage MUST be covered by TODOs.**
- [ ] Helpers and types promote readability and maintainability.

When landing this PR, be sure to make any necessary issue status updates.
